ui/src/app/components/workflows/workflow/jobs/jobs.component.spec.ts

```diff
@@ -16,25 +16,120 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { JobsComponent } from './jobs.component';
+import { FormPart, WorkflowFormPartsModel } from '../../../../models/workflowFormParts.model';
+import { provideMockStore } from '@ngrx/store/testing';
+import { Store } from '@ngrx/store';
+import { WorkflowAddEmptyJob } from '../../../../stores/workflows/workflows.actions';
 
 describe('JobsComponent', () => {
-  // let component: JobsComponent;
-  // let fixture: ComponentFixture<JobsComponent>;
-  //
-  // beforeEach(async(() => {
-  //   TestBed.configureTestingModule({
-  //     declarations: [ JobsComponent ]
-  //   })
-  //   .compileComponents();
-  // }));
-  //
-  // beforeEach(() => {
-  //   fixture = TestBed.createComponent(JobsComponent);
-  //   component = fixture.componentInstance;
-  //   fixture.detectChanges();
-  // });
-  //
-  // it('should create', () => {
-  //   expect(component).toBeTruthy();
-  // });
+  let fixture: ComponentFixture<JobsComponent>;
+  let underTest: JobsComponent;
+
+  const initialAppState = {
+    workflows: {
+      workflowFormParts: new WorkflowFormPartsModel(
+        [],
+        undefined,
+        new FormPart('jobStaticPart', 'jobStaticPart', true, 'jobStaticPart'),
+        undefined,
+        undefined,
+      ),
+      workflowAction: {
+        mode: 'mode',
+        workflowData: {
+          jobs: [
+            {
+              order: 0,
+              job: [{ property: 'jobStaticPart', value: 'value' }],
+            },
+          ],
+        },
+      },
+    },
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      providers: [provideMockStore({ initialState: initialAppState })],
+      declarations: [JobsComponent],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(JobsComponent);
+    underTest = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    expect(underTest).toBeTruthy();
+  });
+
+  it('should after view init set component properties', async(() => {
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      expect(underTest.mode).toBe(initialAppState.workflows.workflowAction.mode);
+      expect(underTest.hiddenJobs).toEqual([]);
+      expect(underTest.staticJobPart).toEqual(initialAppState.workflows.workflowFormParts.staticJobPart);
+      expect(underTest.jobData).toEqual(initialAppState.workflows.workflowAction.workflowData.jobs);
+    });
+  }));
+
+  it('toggleJob() should push job into hiddenJobs when it is not included with false value', async(() => {
```

Comment:
Is this test-case not redundant in presence of the following one?

Reply:
Yes, I wanted to have it there and explicitly say push when its empty